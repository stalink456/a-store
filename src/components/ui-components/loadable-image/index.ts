export { LoadableImage } from './loadable-image';
