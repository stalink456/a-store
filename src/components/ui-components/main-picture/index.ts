export { MainPicture } from "./main-picture";
