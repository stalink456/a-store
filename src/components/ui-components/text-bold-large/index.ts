export { TextBoldLarge } from "./text-bold-large";
