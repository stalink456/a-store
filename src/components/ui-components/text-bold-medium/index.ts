export { TextBoldMedium } from "./text-bold-medium";
