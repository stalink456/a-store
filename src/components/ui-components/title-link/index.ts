export { TitleLink } from './title-link';
