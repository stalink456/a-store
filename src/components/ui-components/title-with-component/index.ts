export { TitleWithComponent } from "./title-with-component";
