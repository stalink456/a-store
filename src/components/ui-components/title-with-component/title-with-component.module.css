.title_with_component {
  margin-bottom: var(--gap-2xl);
}

.title_with_component form {
  margin: var(--gap-2xl);
}