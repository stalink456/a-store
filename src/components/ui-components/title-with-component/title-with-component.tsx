import React from "react";
import { Typography } from "@alfalab/core-components/typography";
import { TitleWithInputType } from "./types";

import styles from "./title-with-component.module.css";

export const TitleWithComponent: React.FC<TitleWithInputType> = React.memo(
  ({ title, children }) => {
    return (
      <div className={styles.title_with_component}>
        <Typography.Title tag="h6" weight="bold" view="small">
          {title}
        </Typography.Title>
        {children}
      </div>
    );
  }
);
