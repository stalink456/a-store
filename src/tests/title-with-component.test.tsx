import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TitleWithComponent } from 'components/ui-components/title-with-component';
import { MaskedInput } from '@alfalab/core-components/masked-input';

describe('TitleWithComponent component', () => {
  it('should render TitleWithComponent component', () => {
    const { baseElement } = render(
      <TitleWithComponent title='Title'>
        <MaskedInput
          placeholder={'Индекс, город, улица, дом, квартира'}
          block
          data-testid='input-with-address'
        />
      </TitleWithComponent>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render TitleWithComponent component with title', async () => {
    render(
      <TitleWithComponent title='Title'>
        <MaskedInput
          placeholder={'Индекс, город, улица, дом, квартира'}
          block
          data-testid='input-with-address'
        />
      </TitleWithComponent>
    );

    expect(await screen.findByText('Title')).toBeInTheDocument();
  });

  it('should render TitleWithComponent component with input some value', async () => {
    render(
      <TitleWithComponent title='Title'>
        <MaskedInput
          placeholder={'Индекс, город, улица, дом, квартира'}
          block
          data-testid='input-with-address'
        />
      </TitleWithComponent>
    );

    fireEvent.input(screen.getByTestId('input-with-address'), {
      target: { value: 'Some text' },
    });

    expect(screen.getByTestId('input-with-address')).toHaveValue('Some text');
  });

  it("shouldn't render TitleWithComponent component with non-existent text", async () => {
    render(
      <TitleWithComponent title='Title'>
        <MaskedInput
          placeholder={'Индекс, город, улица, дом, квартира'}
          block
          data-testid='input-with-address'
        />
      </TitleWithComponent>
    );

    expect(screen.queryByText('non-existent text')).not.toBeInTheDocument();
  });
});
