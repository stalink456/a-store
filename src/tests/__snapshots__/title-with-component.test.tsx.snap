// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleWithComponent component should render TitleWithComponent component 1`] = `
<body>
  <div>
    <div
      class="title_with_component"
    >
      <h6
        class="typography__component_19zgr typography__styrene-small_pxbut typography__bold_pxbut"
      >
        Title
      </h6>
    </div>
  </div>
</body>
`;
